src/main/java/com/neuvector/Scanner.java

```diff
@@ -54,11 +54,23 @@ public static ScanRepoReportData scanRegistry(Registry registry, NVScanner nvSca
         }else{
             String[] credentials = {registry.getLoginPassword(), license};
             if( scanLayers ) {
-                String[] cmdArgs = {"docker", "run", "--name", generateScannerName(), "--rm", "-v", Scanner.SOCKET_MAPPING, "-v", getMountPath(nvScanner), "-e", "SCANNER_REPOSITORY=" + registry.getRepository(), "-e", "SCANNER_TAG=" + registry.getRepositoryTag(), "-e", "SCANNER_LICENSE=" + license, "-e", "SCANNER_REGISTRY=" + registry.getRegistryURL(), "-e", "SCANNER_REGISTRY_USERNAME=" + registry.getLoginUser(), "-e", "SCANNER_REGISTRY_PASSWORD=" + registry.getLoginPassword() , "-e", "SCANNER_SCAN_LAYERS=true", getNVImagePath(nvScanner.getNvScannerImage(), nvScanner.getNvRegistryURL())};
-                reportData = runScan(cmdArgs, nvScanner.getNvMountPath(), credentials);
+                if (registry.getLoginUser() == null && registry.getLoginPassword() == null) {
+                    String[] cmdArgs = {"docker", "run", "--name", generateScannerName(), "--rm", "-v", Scanner.SOCKET_MAPPING, "-v", getMountPath(nvScanner), "-e", "SCANNER_REPOSITORY=" + registry.getRepository(), "-e", "SCANNER_TAG=" + registry.getRepositoryTag(), "-e", "SCANNER_LICENSE=" + license, "-e", "SCANNER_REGISTRY=" + registry.getRegistryURL(), "-e", "SCANNER_SCAN_LAYERS=true", getNVImagePath(nvScanner.getNvScannerImage(), nvScanner.getNvRegistryURL())};
```

Comment:
Looks like a big difference but actually we are only omitting/including `SCANNER_REGISTRY_USERNAME` and `SCANNER_REGISTRY_PASSWORD` if they are provided. We do not provide these environmental variables to the neuvector container if they were not provided.